### PR TITLE
Added CPP Code for Solution-3 (*^▽^*)

### PR DESCRIPTION
**Pull request by Ishwarendra**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [ ] I have tested my code.
- [ ] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [ ] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-712.mdx [hunk 1]
```diff
@@ -160,7 +160,66 @@ For every cow (say cow $N$), we can check which cows have one entry/exit point b
...
+		{
+			// Make a list of all cows that have an entry/exit point
+			// between the entry and exit point of the current cow
+
+			std::set<char> appearances;
+
+			int start = first_found[recordings[i] - 'A'] + 1, end = i;
+			for (int j = start; j < end; j++)
+			{
+				if (appearances.count(recordings[j]))
```
Comment (envyaims) [inline solutions/usaco-712.mdx:200]: i recommend putting brackets around every statement

**Review by envyaims — APPROVED** [review]
looks great! dont forget to add yourself to the author list :)

File: solutions/usaco-712.mdx (hunk 1 above)
Comment (Ishwarendra) [reply solutions/usaco-712.mdx:200]: > looks great! dont forget to add yourself to the author list :)

Hehe.. Thanks for telling

Comment (Ishwarendra) [reply solutions/usaco-712.mdx:200]: > i recommend putting brackets around every statement
Ok I will put them up

**Review by envyaims — APPROVED** [review]
no need to request another review after i approved lol

File: solutions/usaco-712.mdx [hunk 2]
```diff
@@ -160,7 +160,72 @@ For every cow (say cow $N$), we can check which cows have one entry/exit point b
...
+				else
+				{
+					appearances.insert(recordings[j]);
+				}
+			}
+
+			// Add all pairs to the crossings list if not already in it
+			for (int k : appearances)
+			{
+				if (!crossings.count({recordings[i], k}) && !crossings.count({k, recordings[i]}))
```
Comment (SansPapyrus683) [inline solutions/usaco-712.mdx:213]: over 80 chars

File: solutions/usaco-712.mdx [hunk 3]
```diff
@@ -160,7 +160,72 @@ For every cow (say cow $N$), we can check which cows have one entry/exit point b
...
+				{
+					crossings.insert({recordings[i], k});
+				}
+			}
+		}
+	}
+
+	std::cout << crossings.size() << "\n";
+
+	return 0;
```
Comment (SansPapyrus683) [inline solutions/usaco-712.mdx:223]:
```suggestion
```

File: solutions/usaco-712.mdx [hunk 4]
```diff
@@ -160,7 +160,72 @@ For every cow (say cow $N$), we can check which cows have one entry/exit point b
...
+			{
+				if (!crossings.count({recordings[i], k}) && !crossings.count({k, recordings[i]}))
+				{
+					crossings.insert({recordings[i], k});
+				}
+			}
+		}
+	}
+
+	std::cout << crossings.size() << "\n";
```
Comment (SansPapyrus683) [inline solutions/usaco-712.mdx:221]:
```suggestion
	std::cout << crossings.size() << endl;
```

File: solutions/usaco-712.mdx [hunk 5]
```diff
@@ -160,7 +160,72 @@ For every cow (say cow $N$), we can check which cows have one entry/exit point b
...
 
 <LanguageSection>
-	
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+const int TOTAL = 52;
```
Comment (SansPapyrus683) [inline solutions/usaco-712.mdx:169]: quick comment explaining what this means

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-712.mdx (hunk 5 above)
Comment (Ishwarendra) [reply solutions/usaco-712.mdx:169]: Line 167 to 169? 
In question the length of string is fixed (26 * 2 = 56) so I decided to add a const variable representing total number of characters.

File: solutions/usaco-712.mdx (hunk 2 above)
Comment (Ishwarendra) [reply solutions/usaco-712.mdx:213]: splitted into steps ✅

File: solutions/usaco-712.mdx [hunk 6]
```diff
@@ -160,7 +160,72 @@ For every cow (say cow $N$), we can check which cows have one entry/exit point b
 **Time Complexity:** $\mathcal{O}(T^3)$, where $T$ represents the total number of characters in Farmer John's string.
 
 <LanguageSection>
-	
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
```
Comment (Aadit-Ambadkar) [inline solutions/usaco-712.mdx:168]:
```suggestion
using namespace std;

```

File: solutions/usaco-712.mdx [hunk 7]
```diff
@@ -160,7 +160,72 @@ For every cow (say cow $N$), we can check which cows have one entry/exit point b
...
+const int TOTAL = 52;  // Crossing Points' Map Length 
+
+int main()
+{
+	freopen("circlecross.in", "r", stdin);
+	freopen("circlecross.out", "w", stdout);
+
+	std::string recordings;
+	std::cin >> recordings;
```
Comment (Aadit-Ambadkar) [inline solutions/usaco-712.mdx:177]:
```suggestion
	string recordings;
	cin >> recordings;
```

File: solutions/usaco-712.mdx [hunk 8]
```diff
@@ -160,7 +160,72 @@ For every cow (say cow $N$), we can check which cows have one entry/exit point b
...
+
+	std::vector<int> first_found(26, -1);
+	std::set<std::pair<char, char>> crossings;
```
Comment (Aadit-Ambadkar) [inline solutions/usaco-712.mdx:180]:
```suggestion
	vector<int> first_found(26, -1);
	set<pair<char, char>> crossings;
```

File: solutions/usaco-712.mdx [hunk 9]
```diff
@@ -160,7 +160,72 @@ For every cow (say cow $N$), we can check which cows have one entry/exit point b
...
+			// first occurrence of this letter
+			first_found[recordings[i] - 'A'] = i;
+			continue;
+		}
+		else
+		{
+			// keep track of all cows that have an entry/exit point
+			// between the entry and exit point of the current cow
+
+			std::set<char> appearances;
```
Comment (Aadit-Ambadkar) [inline solutions/usaco-712.mdx:195]:
```suggestion
			set<char> appearances;
```

File: solutions/usaco-712.mdx [hunk 10]
```diff
@@ -160,7 +160,72 @@ For every cow (say cow $N$), we can check which cows have one entry/exit point b
...
+				already_present |= crossings.count({k, recordings[i]});
+				if (!already_present)
+				{
+					crossings.insert({recordings[i], k});
+				}
+			}
+		}
+	}
+
+	std::cout << crossings.size() << std::endl;
```
Comment (Aadit-Ambadkar) [inline solutions/usaco-712.mdx:223]:
```suggestion
	cout << crossings.size() << "\n";
```

**Review by Aadit-Ambadkar — CHANGES_REQUESTED** [review]
ahh using namespace std

File: solutions/usaco-712.mdx (hunk 6 above)
Comment (SansPapyrus683) [reply solutions/usaco-712.mdx:168]: hey op, all of these changes are optional, if you wanna qualify the namespace each time it's fine by me

Comment (Ishwarendra) [reply solutions/usaco-712.mdx:168]: f these changes are optional, if you wanna qualify the namespace each time it's fine by me



> hey op, all of these changes are optional, if you wanna qualify the namespace each time it's fine by me

i think it's ok to keep them.

**Review by Ishwarendra — COMMENTED** [review]
No issue with any changes done.

**Review by SansPapyrus683 — APPROVED** [review]

**Pull request merged**